### PR TITLE
Add documentation on Autocluster and Autoheal

### PR DESCRIPTION
**Pull request by emqplus**

### CODE REVIEW

File: source/changes.rst [hunk 1]
```diff
@@ -5,6 +5,90 @@
...
 
+.. _release_2.3-beta.1:
+
+------------------
+Version 2.3-beta.1
+------------------
+
+*Release Date: 2017-07-24*
+
+EMQ R2.3-beta.1 is available now!. 该版本正式支持集群节点自动发现与集群脑裂自动愈合，支持基于IP Multicast、Etcd、Kubernetes等多种策略自动构建集群。
```
Comment (justina111) [inline source/changes.rst]: Should remove Chinese here.

File: source/changes.rst [hunk 2]
```diff
@@ -5,6 +5,90 @@
...
+EMQ R2.3 supports node discovery and autocluster with various strategies:
+
++------------+---------------------------------+
+| Strategy   | Description                     |
++============+=================================+
+| static     | Autocluster by static node list |
++------------+---------------------------------+
+| mcast      | Autocluster by UDP Multicast    |
++------------+---------------------------------+
+| dns        | Autocluster by DNA A Record     |
```
Comment (justina111) [inline source/changes.rst]: Change 'DNA' to 'DNS'

File: source/changes.rst [hunk 3]
```diff
@@ -5,6 +5,90 @@
...
+A down node will be removed from the cluster if autoclean is enabled:
+
+.. code-block:: properties
+
+    cluster.autoclean = 5m
+
+LWM2M Protocol Support
+-----------------------
+
+EMQ-LWM2M is an gatewaty plugin for EMQ，which implemnts most LWM2M features. MQTT client is able to access LWM2M device through emq-lwm2m plugin, by sending a command and reading its response.
```
Comment (justina111) [inline source/changes.rst]: Change 'is an gateway...' to 'is a gateway...'

File: source/changes.rst [hunk 4]
```diff
@@ -5,6 +5,90 @@
...
+
+Debian 9 Package
+----------------
+
+EMQ R2.3 released binary package for Debian 9.
+
+Erlang/OTP R20
+--------------
+
+EMQ R2.3 is compatible with Erlang/OTP R20, and all the binary packages is built on Erlang/OTP R20.
```
Comment (justina111) [inline source/changes.rst]: Change 'is' to 'are' in 'all the binary packages is built...'

File: source/changes.rst [hunk 5]
```diff
@@ -5,6 +5,90 @@
...
+Network Partition and Autoheal
+------------------------------
+
+Enable autoheal of Network Partition by default:
+
+.. code-block:: properties
+
+    cluster.autoheal = on
+
+When network partition occurs, the following steps to heal the cluster if autoheal is enabled:
```
Comment (grutabow) [inline source/changes.rst]: ..., heal the cluster in following steps if autoheal is enabled:

File: source/cluster.rst [hunk 6]
```diff
@@ -214,13 +214,149 @@ Two ways to leave the cluster:
...
+EMQ R2.3 supports node discovery and autocluster with various strategies:
+
++------------+---------------------------------+
+| Strategy   | Description                     |
++============+=================================+
+| static     | Autocluster by static node list |
++------------+---------------------------------+
+| mcast      | Autocluster by UDP Multicast    |
++------------+---------------------------------+
+| dns        | Autocluster by DNA A Record     |
```
Comment (justina111) [inline source/cluster.rst]: Change 'DNA' to 'DNS'

File: source/cluster.rst [hunk 7]
```diff
@@ -214,13 +214,149 @@ Two ways to leave the cluster:
...
+
+    cluster.mcast.ports = 4369,4370
+
+    cluster.mcast.iface = 0.0.0.0
+
+    cluster.mcast.ttl = 255
+
+    cluster.mcast.loop = on
+
+Autocluster by DNA A Record
```
Comment (justina111) [inline source/cluster.rst]: Change 'DNA' to 'DNS'

File: source/cluster.rst (hunk 6 above)
Comment (grutabow) [inline source/cluster.rst]: DNA  ->  DNS

File: source/cluster.rst (hunk 7 above)
Comment (grutabow) [inline source/cluster.rst]: DNA  ->  DNS

File: source/config.rst [hunk 8]
```diff
@@ -60,6 +60,144 @@ OS Environment Variables
...
+EMQ R2.3 supports node discovery and autocluster with various strategies:
+
++------------+---------------------------------+
+| Strategy   | Description                     |
++============+=================================+
+| static     | Autocluster by static node list |
++------------+---------------------------------+
+| mcast      | Autocluster by UDP Multicast    |
++------------+---------------------------------+
+| dns        | Autocluster by DNA A Record     |
```
Comment (justina111) [inline source/config.rst]: Change 'DNA' to 'DNS'

File: source/config.rst [hunk 9]
```diff
@@ -60,6 +60,144 @@ OS Environment Variables
...
+
+    cluster.mcast.ports = 4369,4370
+
+    cluster.mcast.iface = 0.0.0.0
+
+    cluster.mcast.ttl = 255
+
+    cluster.mcast.loop = on
+
+Autocluster by DNA A Record
```
Comment (justina111) [inline source/config.rst]: Change 'DNA' to 'DNS'

**Pull request merged**